internal/keycloakb/onboardingmodule.go

```diff
@@ -0,0 +1,104 @@
+package keycloakb
+
+import (
+	"context"
+	"crypto/rand"
+	b64 "encoding/base64"
+	"encoding/json"
+	"net/url"
+	"time"
+
+	"github.com/cloudtrust/common-service/log"
+	"github.com/cloudtrust/keycloak-bridge/internal/constants"
+	kc "github.com/cloudtrust/keycloak-client"
+)
+
+// TrustIDAuthToken struct
+type TrustIDAuthToken struct {
+	Token     string `json:"token"`
+	CreatedAt int64  `json:"created_at"`
+}
+
+// ToJSON converts TrustIDAuthToken to its JSON representation
+func (t TrustIDAuthToken) ToJSON() string {
+	var authBytes, _ = json.Marshal(t)
+	return string(authBytes)
+}
+
+type onboardingModule struct {
+	keycloakClient OnboardingKeycloakClient
+	keycloakURL    string
+	logger         log.Logger
+}
+
+type OnboardingKeycloakClient interface {
+	ExecuteActionsEmail(accessToken string, realmName string, userID string, actions []string, paramKV ...string) error
+}
+
+type OnboardingModule interface {
```

Comment:
In theory this interface is useless and should never be used. However, we have golint issue if we don't have this interface. I think that's why we did it that way everywhere .... If we would have time, it would be interesting to look what is the way to do it as the interface must be outside in Golang